compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/AIETarget.cpp

```diff
@@ -150,6 +150,39 @@ class AIETargetBackend final : public IREE::HAL::TargetBackend {
     Builder b(context);
     SmallVector<NamedAttribute> configItems;
 
+    uint32_t maxCoreRows, maxCoreCols;
+    switch (options.AMDAIETargetDevice) {
+      case AMDAIEDevice::npu1:
+      case AMDAIEDevice::npu1_4col:
+        maxCoreRows = 4;
+        maxCoreCols = 4;
+        break;
+      case AMDAIEDevice::npu1_1col:
+        maxCoreRows = 4;
+        maxCoreCols = 1;
+        break;
+      case AMDAIEDevice::npu1_2col:
+        maxCoreRows = 4;
+        maxCoreCols = 2;
+        break;
+      case AMDAIEDevice::npu1_3col:
+        maxCoreRows = 4;
+        maxCoreCols = 3;
+        break;
+      case AMDAIEDevice::npu4:
+        maxCoreRows = 4;
+        maxCoreCols = 8;
+        break;
+      default:
+        llvm::errs() << "unhandled NPU partitioning.\n";
```

Comment:
Yes I'd put an `assert(false && "unhandled target device, we must specify the array size here")`

But isn't this info available already from somewhere like https://github.com/nod-ai/iree-amd-aie/blob/7c4b98571d86d7ec25daa20f1d0ac9e2be8a8c05/runtime/src/iree-amd-aie/aie_runtime/iree_aie_runtime.h#L284 ?

Reply:
It's not just printing the message, it will bail out like `assert`. And there are `llvm::errs()` used through out this file.

@newling @jtuyls Yes, we can use `deviceModel.configPtr.NumCols` to get num_rows/cols, but it still need some hardcode adjustments to show number of rows/cols for cores. For example, `npu1_4col` returns 5 cols and 6 rows, so we need to -1 cols and -2 rows for cores. While `npu4` returns 8 cols and 6 rows, so we only need to -2 rows for cores.

Reply:
You can create a new function in the device model `getNumCoreRows` that returns the number of rows with cores and for the 5 columns, we probably just need to change that in device model.

Reply:
For number of columns, we might be able to use `partitionNumCols`? https://github.com/nod-ai/iree-amd-aie/blob/40628f5528fae75d8a51978247355e42ed954588/runtime/src/iree-amd-aie/aie_runtime/iree_aie_runtime.cc#L589

But the value for `npu1` is 5, and we want 4 instead.

Reply:
Okay, I figured out we can use `deviceModel.cols()` to get correct number of cols from the device. It returns 4 cols for `npu1_4col`, and 5 cols for `npu1`. But I think it's okay, we'd never use 5 cols for npu1 device.